django/contrib/admin/static/admin/js/admin/RelatedObjectLookups.js

```diff
@@ -120,7 +120,7 @@
         var selects = django.jQuery(selectsSelector);
         selects.find('option').each(function() {
             if (this.value === objId) {
-                this.innerHTML = newRepr;
+                this.innerText = newRepr;
```

Comment:
Yes, the ordering of the characters can't be entirely random or else the browser may not treat the string as HTML in all cases. I think putting such characters in a primary key isn't common, so it's not a really important fix but can be done as a separate ticket on master only.